Use RESERVE_SEGMENT on SPARC to limit stack below 47bits of address space.

See similar FF patch which provides js/mapfile.SolarisSPARC.

--- thunderbird-153.2.0/comm/mail/app/moz.build
+++ thunderbird-153.2.0/comm/mail/app/moz.build
@@ -126,6 +126,10 @@ CONFIG["MOZ_BRANDING_DIRECTORY"],
         icon,
     )
 
+# Enforce 47bits address space to resulting binary (Bug 577056).
+if CONFIG["OS_ARCH"] == "SunOS" and CONFIG["TARGET_CPU"] == "sparc64":
+    LDFLAGS += ["-Wl,-M/usr/lib/ld/map.vas47"]
+
 if CONFIG["MOZ_ASAN"] or CONFIG["MOZ_DEBUG"]:
     WINCONSOLE = True
 else:
